File: python solutions/645.py
class Solution:
  def findErrorNums(self, nums):
    map = dict() 
    ret = list()  

    for i, num in enumerate(nums): 
      if i+1 not in map.keys(): 
        map[i+1] = 0 


      if num in map.keys(): 
        map[num] += 1 
        if map[num] > 1: 
          ret.insert(0, num)  
      else: 
        map[num] = 1 
    
    for key in map: 
      if not map[key]: 
        ret.append(key) 

    return ret 


  

File: cpp solutions/645.cpp
#include <iostream> 
#include <vector> 
#include <unordered_map> 

class Solution {
public:
    std::vector<int> findErrorNums(std::vector<int>& nums) {
        std::vector<int> ret;  
        std::unordered_map<int, int> hashMap; 

        for(int i = 1; i <= nums.size(); i++){ 
          
          if(!hashMap[i])
            hashMap[i] = 0;  

          if (hashMap[nums[i-1]] >= 1){ 
            hashMap[nums[i-1]] += 1;
            ret.insert(ret.begin(), nums[i-1]); 
          } else { 
            hashMap[nums[i-1]] = 1;
          }
        }


        for(int i = 1; i <= hashMap.size(); i++)
          if(!hashMap[i]) ret.push_back(i);  
      
        return ret;
    }
};
